src/main/resources/project/server/stopDomain.pl

```diff
@@ -212,37 +216,39 @@ sub main {
         }
         else {
             my $error_summary = "Something wrong after stopping all slave host controllers (before stopping master host controller '$master_host').";
-            $error_summary .= "\nExpected is to have only master host controller '$master_host' started at this point, but actual list of started host controllers is: [" . join(", ", @all_hosts_after_all_slaves_shutdown) . "]";
+            $error_summary .= "\nExpected is to have only master host controller '$master_host' started at this point, but actual list of started host controllers is: ["
+                . join(", ", @all_hosts_after_all_slaves_shutdown) . "]";
```

Comment:
Interpolation with concatenation and function call in the same line is not good. In such cased sprintf should be used.

Reply:
done